### PR TITLE
Upgrade to ldk 116 alpha

### PR DESCRIPTION
**Pull request by TonyGiorgio**
Need to test more and manage deps better.

### CODE REVIEW

File: mutiny-core/src/keymanager.rs [hunk 1]
```diff
@@ -156,21 +157,21 @@ impl<S: MutinyStorage> SignerProvider for PhantomKeysManager<S> {
...
+        Ok(wallet
             .get_address(AddressIndex::New)
             .address
-            .script_pubkey()
+            .script_pubkey())
     }
 
-    fn get_shutdown_scriptpubkey(&self) -> ShutdownScript {
+    fn get_shutdown_scriptpubkey(&self) -> Result<ShutdownScript, ()> {
         let mut wallet = self.wallet.wallet.try_write().unwrap();
```
Comment (benthecarman) [inline mutiny-core/src/keymanager.rs:169]: can map this to an err too

File: mutiny-core/src/keymanager.rs [hunk 2]
```diff
@@ -156,21 +157,21 @@ impl<S: MutinyStorage> SignerProvider for PhantomKeysManager<S> {
         self.inner.read_chan_signer(reader)
     }
 
-    fn get_destination_script(&self) -> Script {
+    fn get_destination_script(&self) -> Result<Script, ()> {
         let mut wallet = self.wallet.wallet.try_write().unwrap();
```
Comment (benthecarman) [inline mutiny-core/src/keymanager.rs:161]: should map this as an err

Comment (TonyGiorgio) [reply mutiny-core/src/keymanager.rs:161]: This is a trait we are inheriting and the signature they're specifying.

**Comment by benthecarman** [comment]
you should be able to change this to use update_channel_from_announcement_no_lookup and delete the dummy utxo lookup impl i made

[master/mutiny-core/src/peermanager.rs#L227](https://github.com/MutinyWallet/mutiny-node/blob/master/mutiny-core/src/peermanager.rs?rgh-link-date=2023-07-06T21%3A30%3A38Z#L227)

**Comment by benthecarman** [comment]
can also remove this whole match statement and replace it with `bal.claimable_amount_satoshis`

https://github.com/MutinyWallet/mutiny-node/blob/f1c7ea0400c2a10d3a8a4fc96743a1d1536a498b/mutiny-core/src/nodemanager.rs#L1268

**Comment by TonyGiorgio** [comment]
Added to changes.

**Review by benthecarman — APPROVED** [review]

**Pull request merged**